disperser/apiserver/controller_client.go

```diff
@@ -0,0 +1,123 @@
+package apiserver
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+
+	pbcommon "github.com/Layr-Labs/eigenda/api/grpc/common/v2"
+	controller "github.com/Layr-Labs/eigenda/api/grpc/controller/v1"
+	"github.com/Layr-Labs/eigenda/api/hashing"
+	aws2 "github.com/Layr-Labs/eigenda/common/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// ControllerClient wraps the controller service client and handles signing of requests
+type ControllerClient struct {
+	controllerAddress  string
+	disperserKMSKeyID  string
+	kmsClient          *kms.Client
+	disperserPublicKey *ecdsa.PublicKey
+
+	clientConnection *grpc.ClientConn
+	serviceClient    controller.ControllerServiceClient
+}
+
+// Creates a client for communicating with the controller GRPC server
+func NewControllerClient(
+	ctx context.Context,
+	controllerAddress string,
+	kmsRegion string,
+	kmsEndpoint string,
+	disperserKMSKeyID string,
+) (*ControllerClient, error) {
```

Comment:
Would it make sense to load the key from kms outside this constructor, and pass it in? Not sure how we do it elsewhere, but by loading it externally it will probably be easier to unit test.